### ISSUE TITLE
Support AggregateExceptions?

### ISSUE THREAD

**Issue by DennisDyallo** [AArnott/Xunit.SkippableFact#31]
Any chance for support of AggregateExceptions could be added easily?

**Comment by AArnott** [comment]
Can you clarify? What exactly would you like to se?

**Comment by DennisDyallo** [comment]
Hi @AArnott and thanks for the quick response!
Well, I'm using xUnit 2.6.1 and inheriting IClassFixtures<T> in my tests. The IClassFixture throws an AggregateException instead of the underlying custom exception that I wish to skip the test on receiving. 

I mean I could go [SkippableTheory(typeof(AggregateException))] instead of [SkippableTheory(typeof(CustomException))], but that will Skip the test for the wrong reasons. 

If SkippableFact could detect the exceptions inside of the AggregateExceptions it would be able to determine to Skip the test. I don't know if the xUnit API allows that as things are. 

Any ideas? 

![image](https://github.com/AArnott/Xunit.SkippableFact/assets/3703696/4fe25179-9bb1-4ce7-8f1f-ab36a4e04e92)

**Comment by AArnott** [comment]
Thanks. That helps a lot. Can you confirm that the _only_ exception within `AggregateException` is the skip exception, such that we can skip in only that condition? Or is it important that we skip if any one exception inside the aggregate is a skip exception, even if other failures are also present?

**Comment by DennisDyallo** [comment]
For my usecase, I'd be happy to simply Skip as soon as a predetermined SkipException is detected. That's because as soon as I detect there's a skip exception thrown, I don't care about the rest. 

And to answer your first question: There was actually another exception in the `AggregateException`:
`Xunit.Sdk.TestClassException: The following constructor parameters did not have matching fixture data: MyFixtureClass myFixtureParam. Exception doesn't have a stacktrace`

Seems to come from here: 
https://github.com/xunit/xunit/blob/21e2eef5a46138e09cd90c8a93d644d5c0d3496a/src/xunit.v3.core/Sdk/v3/Runners/TestClassRunner.cs#L68

**Comment by AArnott** [comment]
Would you be interested in contributing the PR?

**Comment by DennisDyallo** [comment]
Yes but what's the plan on .net6, is the upgrade planned? And this package, is eventually merging into xUnit?

**Comment by AArnott** [comment]
There are no plans to merge this package into xunit. But xunit v3 might be adding its own skippable support, which would make this package obsolete.
As for .NET 6, this package already works for .NET 6 targeting test projects.
